enterprise/utils.py

```diff
@@ -1228,19 +1224,20 @@ def delete_data_sharing_consent(course_id, customer_uuid, user_email):
     TieredCache.delete_all_tiers(consent_cache_key)
 
 
-def validate_email_to_link(email, raw_email=None, message_template=None, ignore_existing=False):
+def validate_email_to_link(email, enterprise_customer, raw_email=None, message_template=None, raise_exception=True):
     """
     Validate email to be linked to Enterprise Customer.
 
     Performs two checks:
         * Checks that email is valid
-        * Checks that it is not already linked to any Enterprise Customer
+        * Checks that it is not already linked to the provided Enterprise Customer
 
     Arguments:
         email (str): user email to link
+        enterprise_customer (EnterpriseCustomer): the enterprise customer to link the email to
         raw_email (str): raw value as it was passed by user - used in error message.
         message_template (str): Validation error template string.
-        ignore_existing (bool): If True to skip the check for an existing Enterprise Customer
+        raise_exception (bool): whether to raise an exception when an email is invalidated
```

Comment:
Nit: I kind of like `ignore_existing` better, I think it more accurately represents what this method does.  If `validate_email()` fails, an exception is raised regardless of the value of `raise_exception`.  `raise_exception` only has an impact if there is an existing ECU record.  Maybe call it `raise_on_existing_link` instead, and update the docstring to indicate that this only has an impact if there is an existing ECU for the given email and customer.

Reply:
I feel like ignore_existing wasn't really good name because what it did was skip/raise an exception when an email was already existing. I feel like a name like ignore_existing would make more sense if it actually meant something like "is this email formatted correctly and if it is then return True regardless of whether or not it's currently linked to an enterprise." What this is currently doing is returning True if the email is formatted correctly and linked with an enterprise, and returning False if it's formatted corrected and not linked with an enterprise regardless of what `ignore_existing` is set to.

Overall I think it's named badly and what it returns should be more clear but that's a really large and high impact change to make right now.